src/System.Net.Http/src/System/Net/Http/Managed/ChunkedEncodingReadStream.cs

```diff
@@ -13,85 +14,81 @@ internal sealed partial class HttpConnection
-                try
+                else if (bytesConsumed != line.Count)
                 {
-                    for (int i = 0; i < line.Count; i++)
+                    // There was data after the chunk size, presumably a "chunk extension".
+                    // Allow tabs and spaces and then stop validating once we get to an extension.
+                    int offset = line.Offset + bytesConsumed, end = line.Count - bytesConsumed + line.Offset;
+                    for (int i = offset; i < end; i++)
                     {
-                        char c = (char)line[i];
-                        if ((uint)(c - '0') <= '9' - '0')
+                        char c = (char)line.Array[i];
+                        if (c == ';')
                         {
-                            size = checked(size * 16 + ((ulong)c - '0'));
+                            break;
                         }
-                        else if ((uint)(c - 'a') <= ('f' - 'a'))
-                        {
-                            size = checked(size * 16 + ((ulong)c - 'a' + 10));
-                        }
-                        else if ((uint)(c - 'A') <= ('F' - 'A'))
-                        {
-                            size = checked(size * 16 + ((ulong)c - 'A' + 10));
-                        }
-                        else
+                        else if (c != ' ' && c != '\t') // not called out in the RFC, but WinHTTP allows it
                         {
                             throw new IOException(SR.net_http_invalid_response);
                         }
                     }
                 }
-                catch (OverflowException e)
+
+                _chunkBytesRemaining = chunkSize;
+                if (chunkSize > 0)
                 {
-                    throw new IOException(SR.net_http_invalid_response, e);
+                    return true;
                 }
-                return size;
+
+                // We received a chunk size of 0, which indicates end of response body. 
+                // Read and discard any trailing headers, until we see an empty line.
+                while (!LineIsEmpty(await _connection.ReadNextLineAsync(cancellationToken).ConfigureAwait(false)));
+
+                _connection.ReturnConnectionToPool();
+                _connection = null;
+                return false;
             }
 
-            private async Task ConsumeChunkBytes(ulong bytesConsumed, CancellationToken cancellationToken)
+            private async Task ConsumeChunkBytesAsync(ulong bytesConsumed, CancellationToken cancellationToken)
             {
                 Debug.Assert(bytesConsumed <= _chunkBytesRemaining);
                 _chunkBytesRemaining -= bytesConsumed;
                 if (_chunkBytesRemaining == 0)
                 {
+                    int currentAllowedReadLineBytes = _connection._allowedReadLineBytes;
+                    _connection._allowedReadLineBytes = MaxChunkBytesAllowed;
```

Comment:
I wonder if this should just be 2?  For CRLF

Reply:
I started with that and then figured "eh, doesn't really matter, might as well use the same constant used elsewhere in the file".  I can make it 2, though.

Reply:
Yeah, I don't think it really matters.  As long at there's some limit.